.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,54 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master,  releasebranch_7_8]
+  pull_request:
+    branches: [ master ]
+  schedule:
+    # Check every Saturday at 18:36
+    - cron: '36 18 * * 6'
+
+jobs:
+  analyze:
+    name: ${{ matrix.language }} - Analyze
+    runs-on: ubuntu-20.04
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # C is included in cpp, no separate C language available on CodeQL
+        language: [ 'cpp', 'python' ]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: ${{ matrix.language }}
+
+    - name: Get dependencies
+      run: |
+        sudo apt-get update -y
+        sudo apt-get install -y wget git gawk findutils
+        xargs -a <(awk '! /^ *(#|$)/' ".github/workflows/apt.txt") -r -- \
+            sudo apt-get install -y --no-install-recommends --no-install-suggests
+    - name: Create installation directory
+      run: |
+        mkdir $HOME/install
+    - name: Ensure one core for compilation
+      run: |
+        echo "MAKEFLAGS=-j1" >> $GITHUB_ENV
```

Comment:
I think reason for this in gcc.yml was just that compilation errors are not mixed. This does not apply here. Using all cores is preferred because of speed, so use nproc like in, e.g., flake8.yml or check what is the best way to do this in GH Actions in combination with make.